pulser/sequence.py

```diff
@@ -774,6 +774,11 @@ def _delay(self, duration: int, channel: str) -> None:
         if self.is_parametrized():
             return
 
+        # Delay must not be shorter than the minimal duration for this channel
+        min_duration = self._channels[channel].min_duration
+        if duration < min_duration:
+            duration = min_duration
+
```

Comment:
I would rather this didn't happen when the user calls `delay`, as I don't want to change an explicit input. Could you do this before `_delay` is called within `add` instead (outside of `_delay`, on line 514)?